Description: Rather than using the logic in the best_name functionality.  libnotify should be using the built-in functionality of pidgin. There is already a function purple_buddy_get_contact_alias which returns the correct alias.
Author: Stephen Ostrow
Bug: http://sourceforge.net/tracker/?func=detail&aid=2831741&group_id=144907&atid=760303
Bug-Ubuntu: https://launchpad.net/bugs/408624

Index: pidgin-libnotify-0.14/src/pidgin-libnotify.c
===================================================================
--- pidgin-libnotify-0.14.orig/src/pidgin-libnotify.c	2012-11-16 10:14:37.751591394 +0100
+++ pidgin-libnotify-0.14/src/pidgin-libnotify.c	2012-11-16 10:14:38.695591359 +0100
@@ -131,16 +131,10 @@
 }
 
 /* do NOT g_free() the string returned by this function */
-static gchar *
+const static gchar *
 best_name (PurpleBuddy *buddy)
 {
-	if (buddy->alias) {
-		return buddy->alias;
-	} else if (buddy->server_alias) {
-		return buddy->server_alias;
-	} else {
-		return buddy->name;
-	}
+  return purple_buddy_get_contact_alias(buddy);
 }
 
 static GdkPixbuf *
